### ISSUE TITLE
un-deprecate `write_grads` for `fit`

### ISSUE THREAD

**Issue by IanQS** [tensorflow/tensorflow#31173]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.0-beta
- Are you willing to contribute it (Yes/No): No

**Describe the feature and the current behavior/state.**

```
tb_callback = tf.keras.callbacks.TensorBoard(log_dir=logdir,
                                                 write_grads=True)
```

- ignores the `write_grads` 

One might argue that I can write a custom `train` function then accumulate the gradients myself but some features I'm using cause the custom train function to fail (`tf.feature_columns`) [BUG: [TF 2.0] categorical_column_with_vocabulary_list not usable in custom training loop](https://github.com/tensorflow/tensorflow/issues/30453)

**Will this change the current api? How?**

writes the grads to Tensorboard

**Who will benefit with this feature?**

People who want to visualize the gradients of their model

**Any Other info.**

Why is it getting deprecated?

**Comment by omalleyt12** [comment]
@IanQS thanks for this feature request! Yes, I think we're at a point with 2.0 and TensorBoard in Keras where we can support this now, I'll aim to get this in the within the next week or two

**Comment by du210** [comment]
Hi @omalleyt12, is there any progress on this? Thanks, Deyan

**Comment by IanQS** [comment]
@omalleyt12 sorry to bother you did you manage to get a solution merged in?

**Comment by pendragonn** [comment]
Hello, is there any progress or alternative solution for this?

Thanks!

**Comment by jarednielsen** [comment]
Is this feature on the roadmap? It's critical for our use case.

**Comment by rodyt** [comment]
Would really like this for debugging. Is this going to be included in TF2.2?

**Comment by fzyzcjy** [comment]
This is really wonderful feature. Looking forward to seeing it again!

**Comment by liuxingbaoyu** [comment]
Hello, are there any updates? Thank you!

**Comment by southisland** [comment]
Any updates?

**Comment by hajohajo** [comment]
This would be a really useful feature to have. Are there any updates?

**Comment by jhandsel** [comment]
I would also like this feature reinstated.

**Comment by xianglinyang** [comment]
Need this feature! Pls add this feature back 0.0

**Comment by Entropy4** [comment]
please re-add this feature, it was so useful back in TF 1.x

**Comment by gudmundurjo** [comment]
I would also like to see this feature for TF2.

**Comment by shaikmoeed** [comment]
Is there any update? Will this feature add in the future? Any best alternative for now?

**Comment by kdonbekci** [comment]
Any updates? Would love to see this feature back!

**Comment by KaleabTessera** [comment]
Any workarounds or updates?

**Comment by atyshka** [comment]
@omalleyt12 It's now been almost 2 years. If we want to contribute this from the community can you at least provide some info on why it was deprecated in the first place?

**Comment by rrklearn2020** [comment]
Any updates? or any Any workarounds for "WARNING:tensorflow:`write_grads` will be ignored in TensorFlow 2.0 for the `TensorBoard` Callback."?

**Comment by IanQS** [comment]
![same](https://user-images.githubusercontent.com/6968573/120903776-ffdc7300-c5fc-11eb-98e7-eb308192655c.png)

M E T A

**Comment by LiranPo** [comment]
Any updates on this? Still a very useful feature, and TF2.8 still doesn't have it :( 
@omalleyt12

**Comment by topinfrassi01** [comment]
Still no updates...

**Comment by sachinprasadhs** [comment]
Hi, Could you please migrate from `feature column` to `keras preprocessing`, `feature column` will be deprecated going forward.
For more details refer [Migrate tf.feature_columns to Keras preprocessing layers](https://www.tensorflow.org/guide/migrate/migrating_feature_columns)
Also, refer the list of API's fate under feature column here https://www.tensorflow.org/api_docs/python/tf/feature_column?version=nightly

**Comment by rchao** [comment]
Hello community, in TF2 with Keras, you could [override the training step](https://keras.io/guides/customizing_what_happens_in_fit/), and collect the metrics you would like to plot on TensorBoard in the [returned dict](https://github.com/keras-team/keras/blob/510df69a824ed41e761fedd9a24a9a0b664ebe74/keras/engine/training.py#L1049). With the TensorBoard callback, it would visit the metrics in the `logs` and log them [here](https://github.com/keras-team/keras/blob/510df69a824ed41e761fedd9a24a9a0b664ebe74/keras/callbacks.py#L2871).

Can you give this a try and let us know whether it works? Thanks.

**Comment by rchao** [comment]
Alternatively, you could call `tf.summary.scalar()` directly in the training step, such as what this [guide](https://www.tensorflow.org/tensorboard/scalars_and_keras#cumulative_batch-level_logging) shows in the code snippets.

**Comment by azagajewski** [comment]
Any progress, this was such a useful feature! @gowthamkpr @ymodak
